source-chargebee-native/source_chargebee_native/models.py

```diff
@@ -0,0 +1,302 @@
+from datetime import datetime, UTC, timedelta
+from typing import Generic, TypeVar, Literal, List, ClassVar
+from pydantic import (
+    AwareDatetime,
+    BaseModel,
+    Field,
+    model_validator,
+)
+
+from estuary_cdk.flow import BasicAuth
+from estuary_cdk.capture.common import (
+    BaseDocument,
+    ResourceState,
+    ConnectorState as GenericConnectorState,
+)
+
+
+def default_start_date():
+    dt = datetime.now(tz=UTC) - timedelta(days=30)
+    return dt
+
+
+class ApiKey(BasicAuth):
```

Comment:
This seems strange. In other connectors that only use an API key we do it like this: https://github.com/estuary/connectors/blob/b41fc920848ff69123516e68861c6e969ba9c6fa/source-gladly/source_gladly/models.py#L34-L38

Reply:
Chargebee uses an API Key as the username in Basic Auth and an empty password..

I could probably approach it the was you've linked, but would have to encode like we do here: https://github.com/estuary/connectors/blob/main/estuary-cdk/estuary_cdk/http.py#L184-L189

Reply:
Oh ok. That's kind of odd but I suppose this will work!

Reply:
For follow-up: I do think we should change the endpoint config for the API token. You should be able to have a input for the API token as source-gladly does, and then initialize the `http.token_source` in a similar way, where the `username` input to `BasicAuth` is the API token and the `password` is hard-coded to an empy string: https://github.com/estuary/connectors/blob/230abdbd0d3d02374839da1e8f50cbaab04d51dc/source-gladly/source_gladly/resources.py#L26-L29